pkg/ksym/ksym.go

```diff
@@ -120,13 +127,14 @@ func (c *cache) Resolve(addrs map[uint64]struct{}) (map[uint64]string, error) {
 	// Fast path for when we've seen this symbol before.
 	c.mtx.RLock()
 	for addr := range addrs {
-		sym, ok := c.fastCache[addr]
+		sym, ok := c.cache.GetIfPresent(addr)
 		if !ok {
 			notCached = append(notCached, addr)
 			c.cacheFetch.WithLabelValues("miss").Inc()
 			continue
 		}
-		res[addr] = sym
+		//nolint:forcetypeassert
```

Comment:
nit: We can check if the type assertion is correct and, if not, return an error. This way, it won't panic if we change the type of things we put in the cache. In any case, this is unlikely to happen considering the locality of this, hence this is a nit-pick.

Reply:
Good idea!